### DISCUSSION TITLE
Catching Signals

### DISCUSSION THREAD

**Discussion by briandowns** [Jason2605/Dictu#751] [General]
I've been playing around with catching signals and different ways it can be done and I've hit a wall with each implementation. I wanted to get some feedback around this topic and possible suggestions on a decent path forward.

**Comment by briandowns** [comment]
It'd be cool to do what Python does an insert the hander for the given signal at the next point of execution on the VM but I haven't quite figured out how to do that.

**Comment by Jason2605** [comment]
Do you have some of the attempts on a repo somewhere? I'll take a look at them. Been a while since I've looked at the Dictu codebase admittedly so will need to refresh myself on it so not much use on hints as of yet but I'll try have a think of something!

(sorry for slow replies)

**Comment by briandowns** [comment]
I think I like this the most. Only thing I need to do is get a signal registered and insert the handler for that signal into the VM at the next point of execution.

https://github.com/dictu-lang/Dictu/pull/752

```cs
import Process;
import System;

def hello() {
    print("handler called after catching signal!);
}

{ // main
    Process.signalHandle(Process.SIGINT, hello);
    System.sleep(10);

    // ctrl c here...

    System.exit(0);
}

```
